Framework/NexusGeometry/test/NexusGeometrySaveTest.h

```diff
@@ -0,0 +1,221 @@
+
+      // attribute of dataset with atrtibute name 'attrName' exists if try block
+      // succeeds.
+      H5::Attribute attribute = dataSet.openAttribute(attrName);
+
+      std::string readClass;
+      attribute.read(attribute.getDataType(), readClass);
+
+	  // if value of attribute read from file matches attrval, returns true.
+      return readClass == attrVal; 
+
+    } catch (...) {
+
+	// DataSet could not be opened.
+      return false; 
+    }
+
+  }
+
+private:
+  H5::H5File m_file;
+};
+
+// Gives a clean file destination and removes the file when handle is out of
+// scope.
+class ScopedFileHandle {
+
+public:
+  ScopedFileHandle(const std::string &fileName) {
+
+    const auto temp_dir = boost::filesystem::temp_directory_path();
+    auto temp_full_path = temp_dir;
+    temp_full_path /= fileName;
+
+    // Check proposed location and throw std::invalid argument if file does
+    // not exist. otherwise set m_full_path to location.
+
+    if (boost::filesystem::is_directory(temp_dir)) {
+      m_full_path = temp_full_path;
+
+    } else {
+      throw std::invalid_argument("failed to load temp directory: " +
+                                  temp_dir.generic_string());
+    }
+  }
+
+  std::string fullPath() const { return m_full_path.generic_string(); }
+
+  ~ScopedFileHandle() {
+
+    // file is removed at end of file handle's lifetime
+    if (boost::filesystem::is_regular_file(m_full_path)) {
+      boost::filesystem::remove(m_full_path);
+    }
+  }
+
+private:
+  boost::filesystem::path m_full_path; // full path to file
+};
+
+} // namespace
+
+//---------------------------------------------------------------------
+
+class NexusGeometrySaveTest : public CxxTest::TestSuite {
+
+private:
+  std::pair<std::unique_ptr<Mantid::Geometry::ComponentInfo>,
+            std::unique_ptr<Mantid::Geometry::DetectorInfo>>
+      m_instrument;
+
+public:
+  // This pair of boilerplate methods prevent the suite being created statically
+  // This means the constructor isn't called when running other tests
+  static NexusGeometrySaveTest *createSuite() {
+    return new NexusGeometrySaveTest();
+  }
+  static void destroySuite(NexusGeometrySaveTest *suite) { delete suite; }
+
+  NexusGeometrySaveTest() {
+    auto instrument = ComponentCreationHelper::createMinimalInstrument(
+        Mantid::Kernel::V3D(0, 0, -10), Mantid::Kernel::V3D(0, 0, 0),
+        Mantid::Kernel::V3D(1, 1, 1));
+
+    m_instrument =
+        Mantid::Geometry::InstrumentVisitor::makeWrappers(*instrument);
+  }
+
+  void test_providing_invalid_path_throws() { // deliberately fails
+
+    std::string path = "invalid_path"; // test invalid path
+
+    TS_ASSERT_THROWS(saveInstrument(*m_instrument.first, path),
+                     std::invalid_argument &);
+  }
+
+  void test_progress_reporting() {
+
+    MockProgressBase progressRep;
+    EXPECT_CALL(progressRep, doReport(testing::_)).Times(1);
+    ScopedFileHandle test("testFile.hdf5");
```

Comment:
Because of the no-sync issue above (see comment). It is important that we do not have name-clashes between our temporary files. Please name each test file after the test name to ensure uniqueness.

Reply:
renamed the files as appropriate